lib/SPIRV/libSPIRV/SPIRVModule.h

```diff
@@ -488,6 +488,14 @@ class SPIRVModule {
     return TranslationOpts.isAllowedToUseExtension(RequestedExtension);
   }
 
+  virtual bool hasProgramAddressSpace() const final {
```

Comment:
Don't get the logic behind virtual final here. Why not just use non-virtual method?

Reply:
I didn't get the logic either, I just copy-pasted method header and changed name. Did not even notice that first.

Reply:
So why adding something you don't understand. I would `virtual` and `final` here.